core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -401,9 +401,7 @@ private[spark] class TaskSchedulerImpl(
                 // addresses are the same as that we allocated in taskResourceAssignments since it's
                 // synchronized. We don't remove the exact addresses allocated because the current
                 // approach produces the identical result with less time complexity.
-                availableResources(i).getOrElse(rName,
-                  throw new SparkException(s"Try to acquire resource $rName that doesn't exist."))
```

Comment:
this should never happens since `taskResAssignmentsOpt` is non empty.

Reply:
`taskResAssignmentsOpt` isn't directly related to `availableResources`, can you explain more to help others review?

Reply:
`task.resources` (L399) is from `taskResAssignmentsOpt` and `taskResAssignmentsOpt` is from `availableResources(i)` (see `resourcesMeetTaskRequirements`).

Reply:
yeah this shouldn't happen I think it was there as just in case a bug or someone changed something that would get caught here.  I'm ok with this

Reply:
Agree, this can be removed and simplified as below.